cli/src/action/circuit/mod.rs

```diff
@@ -166,11 +169,37 @@ impl Action for CircuitProposeAction {
                 .map(ToOwned::to_owned)
                 .or_else(|| std::env::var(SPLINTER_REST_API_URL_ENV).ok())
                 .unwrap_or_else(|| DEFAULT_SPLINTER_REST_API_URL.to_string());
-            let key = args.value_of("key").unwrap_or("./splinter.priv");
 
-            let client = SplinterRestClient::new(&url);
+            let key = args.value_of("key");
+
+            let mut builder = SplinterRestClientBuilder::new();
+            builder = builder.with_url(url);
+
+            #[cfg(feature = "splinter-cli-jwt")]
+            {
+                let default_key_path = dirs::home_dir()
+                    .map(|mut p| {
+                        p.push(".splinter/keys");
+                        p
+                    })
+                    .ok_or_else(|| CliError::EnvironmentError("Home directory not found".into()))?;
+                let default_path_string = {
+                    if let Some(path) = default_key_path.to_str() {
+                        path
+                    } else {
+                        return Err(CliError::ActionError(
+                            "Path is not valid unicode".to_string(),
+                        ));
+                    }
+                };
```

Comment:
IMO, this should all be part of the `create_token` function. The only thing that needs to be passed to that function is the `key` argument.

Reply:
Moved this to the function that creates the token